lxd/instance/drivers/driver_lxc.go

```diff
@@ -799,6 +799,11 @@ func (d *lxc) initLXC(config bool) (*liblxc.Container, error) {
 		"/sys/kernel/tracing",
 	}
 
+	// Pass in /dev/zfs to the container if delegation is supported on the system.
+	if storageDrivers.ZFSSupportsDelegation() && shared.PathExists("/dev/zfs") {
```

Comment:
OK. Do we pass a device for btrfs then?

I'm am concerned with exposing more info about the host to the container unnecessarily (as if I am reading this correctly, it would be exposed even if no delegated ZFS volumes were passed to the container), as well as any potential security issues with access to /dev/zfs would be exacerbated by this because containers get access to it whereas before they wouldn't have.

Perhaps a global config option would be a compromise, like `instances.zfs.delegate`?

Reply:
`/dev/zfs` on a normal host is world-writable, so I'm not concerned about a security issue here.

As for extra knowledge, I fail to see how seeing /dev/zfs is any difference from seeing `zfs` in `/proc/self/mountinfo` or looking at the ZFS version and module through `/sys/module/zfs`.

Reply:
Ah yes, I didn't think about /sys/module/zfs

`/proc/self/mountinfo` doesn't show zfs info unless the container is using zfs, but as `/sys/module/zfs` does then I'm fine with passing /dev/zfs in all cases.

Reply:
(I did consider making /dev/zfs conditional on having either the root disk or a custom volume using delegation, but that would require refcounting for volumes being added/removed and it would also prevent datasets external from LXD from being passed as /dev/zfs wouldn't show up in such cases)